### PR TITLE
fix(nuxt): separate routes for different suspense forks

### LINKED ISSUE #6274: route updates before suspense resolves
Currently using `useRoute` (or `$route`) within a page means that some unexpected and counter-intuitive behaviours can occur when changing routes.

For example, content within a page that is dependent on the route can update _before_ the new page swaps in, resulting in a strange flash of outdated data, or a network request that returns an invalid result (for example, when `route.params.slug` goes to undefined). This is more evident the longer it takes to resolve async setups, so when network speed is lower. In some cases that also mean that invalid data was getting cached within `useAsyncData` responses.

The issue is that `<Suspense>` creates two trees of vnodes, and needs to resolve the hidden tree before swapping it in. But it does not freeze reactivity on the route, so the 'visible' page reacts to the change of route before the new route page actually appears on the page.

**Reproduction**: https://stackblitz.com/edit/github-qwr9bd

CleanShot 2022-08-01 at 09 50 46

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves #6274

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR provides the current route directly at the point of suspense, meaning within the context of a suspense fork, the route will not change until suspense has resolved. We also update the default `useRoute` behaviour to mean that the route will not change outside of a page context until suspense has resolved.

**Context**: https://github.com/nuxt/framework/pull/4124, https://github.com/nuxt/framework/issues/4122, https://github.com/nuxt/framework/pull/4092.

---

**Previous behaviour**:

CleanShot 2022-08-01 at 09 50 46

**New behaviour**:

CleanShot 2022-08-01 at 09 51 37

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/pages/runtime/router.ts [hunk 1]
```diff
@@ -76,15 +76,10 @@ export default defineNuxtPlugin(async (nuxtApp) => {
...
-  // https://github.com/vuejs/vue-router-next/blob/master/src/router.ts#L1192-L1200
-  const route = {}
-  for (const key in router.currentRoute.value) {
-    route[key] = computed(() => router.currentRoute.value[key])
-  }
-
   // Allows suspending the route object until page navigation completes
-  const _activeRoute = shallowRef(router.resolve(initialURL) as RouteLocation)
-  const syncCurrentRoute = () => { _activeRoute.value = router.currentRoute.value }
+  // https://github.com/vuejs/vue-router-next/blob/master/src/router.ts#L1192-L1200
```
Comment (posva) [inline packages/nuxt/src/pages/runtime/router.ts:80]: what is the correct line at https://github.com/vuejs/router/blob/main/packages/router/src/router.ts#L1192-L1200?

Comment (danielroe) [reply packages/nuxt/src/pages/runtime/router.ts]: Updated to https://github.com/vuejs/router/blob/main/packages/router/src/router.ts#L1225-L1233

**Review by Atinux — APPROVED** [review]
Looks good to me ❤️

**Review by pi0 — APPROVED** [review]
Tested locally. All seems great!

**Pull request merged**